collectors/metrics/pkg/metricfamily/label.go

```diff
@@ -53,18 +55,53 @@ func appendLabels(
 	existing []*clientmodel.LabelPair,
 	overrides map[string]*clientmodel.LabelPair) []*clientmodel.LabelPair {
 	var found []string
+
+	// remove blank names and values
+	var withoutEmpties []*clientmodel.LabelPair = make([]*clientmodel.LabelPair, 0)
+
+	for i, pair := range existing {
+		name := pair.GetName()
+		value := pair.GetValue()
+		// remove any name == "" or value == nil
+		if name != "" && value != "" {
+			withoutEmpties = append(withoutEmpties, existing[i])
+		}
+	}
+	existing = withoutEmpties
+
+	// override matching existing labels
 	for i, pair := range existing {
 		name := pair.GetName()
 		if value, ok := overrides[name]; ok {
 			existing[i] = value
 			found = append(found, name)
 		}
 	}
+
+	// append any overrides that didn't already exist
 	for k, v := range overrides {
-		if !contains(found, k) {
-			existing = append(existing, v)
+		// only append real names and values
+		// don't append anything overwritten above
+		if k != "" && v.GetValue() != "" && !contains(found, k) {
+			existing = insertLexicographicallyByName(existing, v)
 		}
 	}
+	sort.SliceStable(existing, func(i, j int) bool {
```

Comment:
do we need to sort all labels here?

Reply:
ah, good point. Probably not, I will test and update here

Reply:
Turns out removing that breaks it. I've gone ahead and just lexicographically inserted instead of appending further up in the code to resolve this